### PR TITLE
[enhancement](Nereids) remove stats derivation in CostAndEnforce job

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

1. remove stats derivation in CostAndEnforce job
2. enforce valid for each stats after estimating

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Comment by englefly** [comment]
how about tpcds q6 performance?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/Statistics.java [hunk 1]
```diff
@@ -123,6 +123,24 @@ public void fix(double newRowCount, double originRowCount) {
         }
     }
 
+    public void enforceValid() {
+        for (Entry<Expression, ColumnStatistic> entry : expressionToColumnStats.entrySet()) {
+            ColumnStatistic columnStatistic = entry.getValue();
+            if (!checkColumnStatsValid(columnStatistic)) {
+                ColumnStatisticBuilder columnStatisticBuilder = new ColumnStatisticBuilder(columnStatistic);
+                columnStatisticBuilder.setNdv(Math.min(columnStatistic.ndv, rowCount));
+                columnStatisticBuilder.setNumNulls(Math.min(columnStatistic.numNulls, rowCount));
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/Statistics.java:132]: numNulls <= rowCount - ndv

**Comment by XieJiann** [comment]
run buildall

**Pull request closed without merging**